internal/mode/static/nginx/config/servers_template.go

```diff
@@ -60,6 +87,20 @@ server {
     include {{ $i.Name }};
         {{- end }}
 
+    {{- if and ($.RewriteClientIP.ProxyProtocol) ($s.IsSocket)}}
+	set_real_ip_from unix:;
+  	{{- else if (not $s.IsSocket)}}
+		{{- range $cidr := $.RewriteClientIP.RealIPFrom }}
+    set_real_ip_from {{ $cidr }};
+        {{- end}}
+	{{ end }}
+        {{- if and ($.RewriteClientIP.RealIPHeader) (not $s.IsSocket)}}
+    real_ip_header {{ $.RewriteClientIP.RealIPHeader }};
+        {{- end}}
+        {{- if and ($.RewriteClientIP.Recursive) (not $s.IsSocket)}}
+    real_ip_recursive on;
+        {{ end }}
+
```

Comment:
Ditto

Reply:
I am wondering how we could do this? 

Each directive set is slightly different based on if the server block is a socket or not and rules are different for stream. 

Do you recommend I associate these settings for each server block? 

I am open to suggestions

Reply:
Your server struct could have fields for all three new directives shown here, and then in the Go code you set these values based on if the server is Socket or not, etc, since you can check that value in the Go code. Then in the template you just include that field if it's set.